fecgov/fec-cms
# Feature/2802 aggregate totals widget

## Summary
Continued development of the aggregate totals box(es).

- Resolves #2802 

## Impacted areas of the application
- Moved widget files from `modules` directories (CSS and JS) into `widgets` directories.
- To get a bare-minimum html page for inside iframes that house only a widget, I added a view specifically for widgets and created a new widget layout.
- Significant progress toward a final version of this widget and framework set up for successive units.

## Screenshots
A shot of a page with a number of different implementations:
![image](https://user-images.githubusercontent.com/26720877/57402160-af3b8700-71a4-11e9-8edf-bdaf1715a9f6.png)

## Related PRs

None

## How to test

- Set env var `FEC_FEATURE_AGGR_TOTS` to positive/true.
- Pull the branch.
- `npm run build`
- If you're getting a CORS warning, you'll need to restart your browser with security turned off.
     - Close Chrome.
     - Run `open -n -a /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --args --user-data-dir="/tmp/chrome_dev_test" --disable-web-security`.
- Go to Wagtail admin, settings > redirects, ADD REDIRECT
     - Redirect from `/widgets/aggregate-totals.js`
     - Redirect to: `http://127.0.0.1:8000/static/js/widgets/aggregate-totals.js`
- Check the page(s) I put together to test many variations of the implementation (127.0.0.1 is important—Chrome seems to like it better than localhost for CORS).
     - http://127.0.0.1:8000/data/spending-bythenumbers/?office=S&election_year=1980
     - http://127.0.0.1:8000/data/raising-bythenumbers/?office=P&election_year=2008
- Compare the grey bars with the units under them—that should explain their initial state/values.
- Change the values of the various pull-downs.
- Check the console to confirm displayed data matches what the API returned.
- Change the URL params however you like. Each unit has a note above it indicating how it should be behaving.
- Feel free to change the data-props for the various <script> in bythenumbers.jinja.
- Restart your browser like normal to re-activate security.

## Known Issues

I wanted to get this in front of you while I finish up the last little bit
- need to polish the layout of a couple widths/layouts
- need to address the TODOs
- need to finish the comments
- will request design approval after responsive issues are addressed
- waiting for final PM and Content approval
____
